compiler/src/yul/mappers/expressions.rs

```diff
@@ -444,6 +464,14 @@ pub fn nonce_to_ptr(nonce: usize) -> yul::Expression {
     literal_expression! { (ptr) }
 }
 
+/// Converts a storage nonce into a pointer based on the keccak256 hash
+pub fn nonce_with_offset_to_ptr(nonce: usize, offset: usize) -> yul::Expression {
+    let ptr = get_full_signature(nonce.to_string().as_bytes());
+    let ptr = literal_expression! { (ptr) };
+    let offset = literal_expression! { (offset) };
+    expression! { (add([ptr], [offset])) }
```

Comment:
@g-r-a-n-t Not sure if this is still the right thing to do after your recent findings about how to deal with storage.  

But if it is, we should probably calculate the `ptr` manually to get rid of the `add(ptr, offset)`. I thought that this is something that the optimizer would do for us but I ran with `--optimize` and it isn't optimizing the `add` away.

Reply:
this is fine for now

Reply:
:+1: I will create an issue to track this optimization though